polkadot/node/subsystem-types/src/messages.rs

```diff
@@ -161,6 +162,7 @@ pub enum CandidateValidationMessage {
 		ValidationCode,
 		CandidateReceipt,
 		Arc<PoV>,
+		ExecutorParams,
```

Comment:
What was the reasoning to go for full `ExecutorParams`? Didn't we agree to pass in the `SessionIndex` which is known by the sending subsystem already, while now everybody has to fetch `ExecutorParams`.

Once you have the `SessionIndex` you can have a reliable way of fetching anything session related, by using some currently active leaf. Threading `ExecutorParams` everywhere seems a bit involved.

Reply:
Well, I'd say I'm trying to fix what we've broken here. As it was explained by @pepyakin, `ValidateFromExhaustive` should be, obviously, exhaustive. It shouldn't collect any data; all the data needed for validation should be caller-provided. I've never seen a clear explanation of that architectural decision (hopefully Sergey could come out with one himself?), and I didn't oppose breaking it in the original executor parametrization PR, but when this issue came up, I thought it would be good to restore the status quo.

Another thought is that dispute and approval subsystems cache `SessionInfo`s anyway, and `ExecutorParams` is a sibling of `SessionInfo`. It seemed logical to cache both in a single place and provide that cached data to the validation subsys when needed.